include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -0,0 +1,126 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_SYCL_BACKEND_IMPL_H
+#define _ONEDPL_SYCL_BACKEND_IMPL_H
+
+#include <CL/sycl.hpp>
+#include "oneapi/dpl/internal/dynamic_selection_traits.h"
+
+#include "oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h"
+
+#include <chrono>
+#include <vector>
+#include <memory>
+#include <utility>
+
+namespace oneapi {
+namespace dpl {
+namespace experimental {
+
+  class sycl_backend {
+  public:
+    using resource_type = sycl::queue;
+    using wait_type = sycl::event;
+    using execution_resource_t = resource_type;
+    using resource_container_t = std::vector<execution_resource_t>;
+
+  private:
+
+    class async_waiter {
+      sycl::event e_;
+      public:
+        async_waiter(sycl::event e) : e_(e) {}
+        sycl::event unwrap() { return e_; }
+        void wait() { e_.wait(); }
+    };
+
+    class submission_group {
+      resource_container_t resources_;
+    public:
+      submission_group(const resource_container_t& v) : resources_(v) { }
+
+      void wait() {
+        for (auto& r : resources_) {
+          unwrap(r).wait();
+        }
+      }
+    };
+
+  public:
+
+    sycl_backend(const sycl_backend& v) = delete;
+
+    sycl_backend() {
+      initialize_default_resources();
+      sgroup_ptr_ = std::make_unique<submission_group>(global_rank_);
+    }
+
+    template<typename NativeUniverseVector>
+    sycl_backend(const NativeUniverseVector& v) {
+      for (auto e : v) {
+        global_rank_.push_back(e);
```

Comment:
Should we reserve required size for ```global_rank_.push_back(e);``` vector before loop?

Reply:
Fixed